python/ray/serve/api.py

```diff
@@ -672,35 +672,33 @@ def connect() -> Client:
     return Client(controller, controller_name, detached=True)
 
 
-def get_current_backend_tag() -> BackendTag:
-    """When called from within a backend, return its backend tag.
+@dataclass
+class BackendContext:
+    """Stores data for Serve API calls from within the user's backend code."""
+    backend_tag: BackendTag
+    replica_tag: ReplicaTag
 
-    Raises:
-        RayServeException if not called from within a Ray Serve backend.
-    """
-    if _INTERNAL_REPLICA_CONTEXT is None:
-        raise RayServeException("`serve.get_current_backend_tag()`"
-                                "may only be called from within a"
-                                "Ray Serve backend.")
-    else:
-        return _INTERNAL_REPLICA_CONTEXT.backend_tag
 
+def get_backend_context():
```

Comment:
@edoakes thoughts on this API?

Reply:
Just for context, I based it off the API for get_runtime_context() in Ray Core: https://docs.ray.io/en/master/package-ref.html#runtime-context-apis.